iet.so/src/iet_worker.cpp

```diff
@@ -114,73 +114,66 @@ IETWorker::IETWorker():endtest(false) {
-
-}
-
-void IETWorker::blasThread(int gpuIdx,  uint64_t matrix_size, std::string  iet_ops_type, 
-    bool start, uint64_t run_duration_ms, int transa, int transb, float alpha, float beta,
-    int iet_lda_offset, int iet_ldb_offset, int iet_ldc_offset, int iet_ldd_offset) {
+void IETWorker::computeThread(void) {
 
     std::chrono::time_point<std::chrono::system_clock> iet_start_time, iet_end_time;
-    double timetakenforoneiteration;
-    double gflops_interval;
-    double duration;
-    uint64_t gem_ops;
+    double duration = 0;
     std::unique_ptr<rvs_blas> gpu_blas;
-    rvs_blas *free_gpublas;
-    string msg;
+    uint64_t _matrix_size_a;
+    uint64_t _matrix_size_b;
+    uint64_t _matrix_size_c;
+
+    if(matrix_size_a && matrix_size_b && matrix_size_c) {
 
-    duration = 0;
-    gem_ops = 0;
-    // setup rvsBlas
-    gpu_blas = std::unique_ptr<rvs_blas>(new rvs_blas(gpuIdx,  matrix_size,  matrix_size,  matrix_size, "default", transa, transb, alpha, beta,
-          iet_lda_offset, iet_ldb_offset, iet_ldc_offset, iet_ldd_offset, iet_ops_type, ""));
+      _matrix_size_a = matrix_size_a;
+      _matrix_size_b = matrix_size_b;
+      _matrix_size_c = matrix_size_c;
+    }
+    else {
+      _matrix_size_a = matrix_size;
+      _matrix_size_b = matrix_size;
+      _matrix_size_c = matrix_size;
+    }
+
+    // setup rvsblas instance
+    gpu_blas = std::unique_ptr<rvs_blas>(new rvs_blas(gpu_device_index, _matrix_size_a, _matrix_size_b, _matrix_size_c, matrix_init, 
+          iet_trans_a, iet_trans_b, iet_alpha_val, iet_beta_val, iet_lda_offset, iet_ldb_offset, iet_ldc_offset, iet_ldd_offset,
+          iet_ops_type, iet_data_type));
 
     //Genreate random matrix data
     gpu_blas->generate_random_matrix_data();
 
     //Copy data to GPU
-    gpu_blas->copy_data_to_gpu(iet_ops_type);
+    gpu_blas->copy_data_to_gpu();
 
     iet_start_time = std::chrono::system_clock::now();
 
     //Hit the GPU with load to increase temperature
     while ((duration < run_duration_ms) && (endtest == false)) {
 
-        //call the gemm blas
-        gpu_blas->run_blass_gemm(iet_ops_type);
-
-        // Waits for GEMM operation to complete
-        if(!gpu_blas->is_gemm_op_complete())
-          continue;
+      for (uint64_t i = 0; i < iet_hot_calls; i++) {
 
-        //get the end time
-        iet_end_time = std::chrono::system_clock::now();
-        //Duration in the call
-        duration = time_diff(iet_end_time, iet_start_time);
-
-        //Converting microseconds to seconds
-        timetakenforoneiteration = duration/1e6;
-        //calculating Gemm count
-        gflops_interval = gpu_blas->gemm_gflop_count()/timetakenforoneiteration;
-        //Print the gflops interval
-        log_interval_gflops(gflops_interval);
-
-        // check end test to avoid unnecessary sleep
-        if (endtest)
-            break;
+        // run GEMM operation
+        if(!gpu_blas->run_blas_gemm()) {
+          endtest = true;
+          break;
```

Comment:
this breaks only from inner loop, is this what you want, if blas fails ? or do we sensibly break out of while loop

Reply:
Since we are setting endtest=true, it will exit the outer loop too.